variation-commons-mongodb/src/main/java/uk/ac/ebi/eva/commons/mongodb/repositories/VariantRepository.java

```diff
@@ -59,5 +62,5 @@
                                                                  List<String> studyIds);
 
     @Query(value = "{'chr': ?0, 'files.sid': {$in : ?1}}}")
-    VariantMongo findOneByChromosomeAndStudyInSorted(String chr, List<String> studyIds, Sort sort);
+    List<VariantMongo> findAllByChromosomeAndStudyInSorted(String chr, List<String> studyIds, Sort sort);
```

Comment:
why changing the query from findOne to findAll? 

Unless you know that this operation is used in application projects, and after seeing the usage in the tests, I think it makes sense to keep as findOne as it was. I think this operation was used in places like the htsget/vcf-dumper where we just needed to know the 2 variants with the lowest and highest position (respectively) in the chromosome. Returning a list (not even a stream!) with all the variants in the chr seems inefficient and not used anyway.

Reply:
> Unless you know that this operation is used in application projects, and after seeing the usage in the tests, I think it makes sense to keep as findOne as it was.
Due to the changes in CrudRepository interface, we can't retain the name findOne if multiple results are returned by the query because it throws an IncorrectResultSizeDataAccessException. 

> Returning a list (not even a stream!) with all the variants in the chr seems inefficient and not used anyway.

Gosh, this was clearly an oversight! Will rectify this...

>I think this operation was used in places like the htsget/vcf-dumper where we just needed to know the 2 variants with the lowest and highest position (respectively) in the chromosome.

I did look for usages with a [global search in Github](https://github.com/search?q=findOneByChromosomeAndStudyInSorted&type=Code) but could not find usages beyond variation-commons.